apps/smart-contracts/token/test/PPO.test.ts

```diff
@@ -153,4 +154,109 @@ describe('=> PPO', () => {
       expect(await ppo.balanceOf(user1.address)).to.eq(MAX_UINT256)
     })
   })
+
+  describe('# burn', () => {
+    beforeEach(async () => {
+      await setupPPO()
+      await ppo.connect(owner).mint(user1.address, 10)
+    })
+
+    it('reverts if amount > balance', async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await expect(ppo.connect(user1).burn(callerPPOBalanceBefore.add(1))).to.revertedWith(
+        'ERC20: burn amount exceeds balance'
+      )
+    })
+
+    it("doesn't decrease caller balance if amount = 0", async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await ppo.connect(user1).burn(0)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(callerPPOBalanceBefore)
+    })
+
+    it('decreases caller balance if amount < balance', async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await ppo.connect(user1).burn(callerPPOBalanceBefore.sub(1))
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(1)
+    })
+
+    it('decreases caller balance if amount = balance', async () => {
+      const callerPPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await ppo.connect(user1).burn(callerPPOBalanceBefore)
+
+      expect(await ppo.balanceOf(user1.address)).to.eq(0)
+    })
+  })
+
+  describe('# burnFrom', () => {
+    beforeEach(async () => {
+      await setupPPO()
+      await ppo.connect(owner).mint(user1.address, 10)
+    })
+
+    it('reverts if burn from zero address', async () => {
+      await expect(ppo.connect(user1).burnFrom(ZERO_ADDRESS, 1)).revertedWith(
+        'ERC20: burn amount exceeds allowance'
+      )
+    })
+
+    it('reverts if caller not approved', async () => {
+      const user1PPOBalanceBefore = await ppo.balanceOf(user1.address)
+
+      await expect(
+        ppo.connect(user2).burnFrom(user1.address, user1PPOBalanceBefore.sub(1))
+      ).to.revertedWith('ERC20: burn amount exceeds allowance')
+    })
+
+    it('reverts if caller approved and amount > allowance', async () => {
+      const user1PPOBalanceBefore = await ppo.balanceOf(user1.address)
+      await ppo.connect(user1).approve(user2.address, user1PPOBalanceBefore.sub(1))
+
+      await expect(
+        ppo.connect(user2).burnFrom(user1.address, user1PPOBalanceBefore)
+      ).to.revertedWith('ERC20: burn amount exceeds allowance')
+    })
+
+    it('reverts if caller approved and amount > balance', async () => {
```

Comment:
```suggestion
    it('reverts if allowance > amount and amount > balance', async () => {
```
Once again, doesn't really make sense to include "caller approved" as a descriptor.
What you are more specifically saying is that the owner's allowance is sufficient, but their balance is insufficient.

Reply:
nit:
```suggestion
    it('reverts if amount > balance', async () => {
```
don't need the `amount < allowance` part - that's implied already since we've already tested the revert cases above